crates/bevy_sprite/src/texture_atlas_builder.rs

```diff
@@ -86,8 +86,9 @@ impl<'a> TextureAtlasBuilder<'a> {
     ///
     /// Optionally an asset id can be passed that can later be used with the texture layout to retrieve the index of this texture.
     /// The insertion order will reflect the index of the added texture in the finished texture atlas.
-    pub fn add_texture(&mut self, image_id: Option<AssetId<Image>>, texture: &'a Image) {
+    pub fn with_texture(mut self, image_id: Option<AssetId<Image>>, texture: &'a Image) -> Self {
```

Comment:
I'm okay with changing `finish`, but doesn't `add_texture` feels better here? We are pushing an image to the atlas, not replacing it, after all.